hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/helpers/KafkaOffsetGen.java

```diff
@@ -182,6 +187,10 @@ public KafkaOffsetGen(TypedProperties props) {
               .map(x -> new TopicPartition(x.topic(), x.partition())).collect(Collectors.toSet());
 
       // Determine the offset ranges to read from
+      if (kafkaCheckpointTimestamp != null) {
+        lastCheckpointStr = Option.of(getOffsetsByTimestamp(consumer, partitionInfoList, topicName, Long.parseLong(kafkaCheckpointTimestamp)));
+      }
+
       if (lastCheckpointStr.isPresent() && !lastCheckpointStr.get().isEmpty()) {
```

Comment:
Here we can not simply over write `lastCheckpointStr`. if user configed `Config.KAFKA_CHECKPOINT_TIMESTAMP`,  hudi will always consume from `Config.KAFKA_CHECKPOINT_TIMESTAMP` and can not moving on, right ?

Reply:
I deal with it now